cmd/tether/attach_linux.go

```diff
@@ -33,6 +33,48 @@ import (
 
 var backchannelMode = os.ModePerm
 
+func setTerminalSpeed(fd uintptr) error {
+	var current struct {
+		termios syscall.Termios
+	}
+
+	// get the current state
+	if _, _, errno := syscall.Syscall6(syscall.SYS_IOCTL,
+		uintptr(fd),
+		syscall.TCGETS,
+		uintptr(unsafe.Pointer(&current.termios)),
+		0,
+		0,
+		0,
+	); errno != 0 {
+		return errno
+	}
+
+	// copy it as the future
+	future := current.termios
+
+	// unset 9600 bps
+	future.Cflag &^= syscall.B9600
+	// set them to 115200 bps
+	future.Cflag |= syscall.B115200
+	future.Ispeed = syscall.B115200
+	future.Ospeed = syscall.B115200
```

Comment:
there were two ways.
1. I measured transfer speed between two points by implementing my own read/write wrapper - it is not precise, however I was getting speeds like 30-50KB/s for terminal sessions while running midnight commander or vmware/powercli - these are doing massive data updates on screen.
2. dd if=/dev/zere of=/.tether/<serial_port>